src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -561,6 +561,14 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             if (repeatCount==0) {
                 return this;
             }
+
+            // this is where we can check if the repeatCount will put us over m_MaxCapacity
+            // We are doing the check here to prevent the corruption of the StringBuilder.
+            int newLength = Length + repeatCount;
+            if (newLength > m_MaxCapacity || newLength < repeatCount) {
+                throw new ArgumentOutOfRangeException("repeatCount", Environment.GetResourceString("ArgumentOutOfRange_NegativeCount"));
```

Comment:
Is this the right resource string?  It maps to "ArgumentOutOfRange_NegativeCount", but repeatCount < 0 was already checked earlier.  This check is really about catching overflow.